File: TaskyJ.Globals.Data/DataObjects/DBRefreshTokenJ.cs
using System;
using System.Runtime.Serialization;

namespace TaskyJ.Globals.Data.DataObjects
{
    public class DBRefreshTokenJ : BaseEntity
    {
        [DataMember]
        public string Token { get; set; }
        [DataMember]
        public DateTime Expires { get; set; }
        [DataMember]
        public DateTime Created { get; set; }
        [DataMember]
        public string CreatedByIp { get; set; }
        [DataMember]
        public DateTime? Revoked { get; set; }
        [DataMember]
        public string RevokedByIp { get; set; }
        [DataMember]
        public string ReplacedByToken { get; set; }

        public bool IsExpired => DateTime.UtcNow >= Expires;
        public bool IsActive => Revoked == null && !IsExpired;

        public override string ToString()
        {
            return Token;
        }

        public override bool Equals(BaseEntity source)
        {
            if (source is DBRefreshTokenJ sourcej)
            {
                //TODO: complete equals method
                return base.Equals(sourcej);
            }
            else
            {
                return base.Equals(source);
            }
        }
    }
}

File: TaskyJ.Interface.Angular/ClientApp/src/app/models/DBRefreshTokenJ.ts
import { BaseEntity } from './BaseEntity';

export class DBRefreshTokenJ implements BaseEntity {
  id: number;
  token: string = null;
  expires: Date = null;
  created: Date = null;
  createdByIp: string = null;
  revoked: Date = null;
  revokedByIp: string = null;
  replacedByToken: string = null;

  /*
          public bool IsExpired => DateTime.UtcNow >= Expires;
          public bool IsActive => Revoked == null && !IsExpired;
  */

  public Equals(source: DBRefreshTokenJ): boolean {
    return this.id == source.id;
  }

  public CopyFrom(source: DBRefreshTokenJ): void {
    this.id = source.id;
  }
}
